File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        map={}
        ma=0
        checked=[]
        for i in nums:
            map[i]=True
        for i in nums:
            temp=i-1
            if(temp in map):
                pass
            
            else:
                count=0
                while(i  in map):
                    i+=1
                    count+=1
                
                ma=max(ma,count)
                    
        return ma

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int ans=0;
        HashSet<Integer> set = new HashSet<>();
        for(int n: nums){
            set.add(n);
        }
        for(int num: set){
            if(!set.contains(num-1)){
                int temp=0;
                while(set.contains(num)){
                    temp++;
                    num++;
                }
                ans=Math.max(ans, temp);
            }
        }
        return ans;
    }
}